app/js/reducers/artifacts.js

```diff
@@ -23,8 +23,11 @@ const artifacts = (state = initialState, action) => {
         });
         return state.set(action.artifact.id, nextState);
     }
-    case 'DELETE_ARTIFACT':
-        return state.delete(action.id);
+    case 'DELETE_ARTIFACT': {
+        // TODO, make it actually do something
+        const newState = ImmutableMap(state);
+        return newState.delete(action.id).toJS();
```

Comment:
We do have support for `es6` list spread `[...list, 4, 5]` and `es7` object spread `{...obj, x:4, y:5}` so it wouldn't be a huge deal if we didn't use ImmutableJS.